commands/promote.js

```diff
@@ -61,6 +63,45 @@ function * run (context, heroku) {
       }
     })
   }))
+
+  let releasePhase = (yield heroku.get(`/apps/${app}/formation`))
+    .find((formation) => formation.type === 'release')
+
+  if (releasePhase) {
+    yield cli.action('Checking release phase', co(function * () {
+      let releases = (yield heroku.get(`/apps/${app}/releases`))
+          .sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at))
+      let attach = releases.find((release) => release.description.includes('Attach DATABASE'))
+      let detach = releases.find((release) => release.description.includes('Detach DATABASE'))
+
+      if (!attach || !detach) {
+        throw new Error(`Unable to find Detach and Attach DATABASE releases ${cli.color.app(app)}`)
+      }
+
+      let [attachId, detachId] = [attach.id, detach.id]
+      while (true) {
+        let attach = yield heroku.get(`/apps/${app}/releases/${attachId}`)
+        if (attach && attach.status === 'succeeded') {
+          let msg = 'pg:promote succeeded.'
+          let detach = yield heroku.get(`/apps/${app}/releases/${detachId}`)
+          if (detach && detach.status === 'failed') {
+            msg += ` It is safe to ignore the failed ${detach.description} release.`
+          }
+          return cli.action.done(msg)
+        } else if (attach && attach.status === 'failed') {
+          let msg = `pg:promote failed because ${attach.description} release was unsuccessful. Your application is currently running `
+          let detach = yield heroku.get(`/apps/${app}/releases/${detachId}`)
+          if (detach && detach.status === 'succeeded') {
+            msg += 'without an attached DATABASE_URL.'
+          } else {
+            msg += `with ${current.addon.name} attached as DATABASE_URL`
+          }
+          return cli.action.done(msg)
+        }
+        yield new Promise((resolve) => setTimeout(resolve, 5000))
```

Comment:
Release phase can (used to be able to?) run for up to 24h, so this could be a while with no feedback. Not that we can really do much.

Reply:
good point. I added a 15 minute timeout.

Reply:
uhmm actually does it mean that `pg:promote` command won't finish for up to 15 mins potentially?

Reply:
Yes #thanksreleasephase